plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4672,22 +4714,64 @@ public void adjust(Mechanism mechanism) {
         // @tags
         // <LocationTag.sign_contents>
         // -->
-        if (mechanism.matches("sign_contents") && getBlockState() instanceof Sign) {
-            Sign state = (Sign) getBlockState();
-            for (int i = 0; i < 4; i++) {
-                PaperAPITools.instance.setSignLine(state, i, "");
-            }
-            ListTag list = mechanism.valueAsType(ListTag.class);
-            CoreUtilities.fixNewLinesToListSeparation(list);
-            if (list.size() > 4) {
-                mechanism.echoError("Sign can only hold four lines!");
+        if (mechanism.matches("sign_contents")) {
+            BlockState state = getBlockState();
+            if (!(state instanceof Sign sign)) {
+                mechanism.echoError("'sign_contents' mechanism can only be called on Sign blocks.");
+            } else {
+                for (int i = 0; i < 4; i++) {
+                    PaperAPITools.instance.setSignLine(sign, i, "");
+                }
+                ListTag list = mechanism.valueAsType(ListTag.class);
+                CoreUtilities.fixNewLinesToListSeparation(list);
+                if (list.size() > 4) {
+                    mechanism.echoError("Sign can only hold four lines!");
+                } else {
+                    for (int i = 0; i < list.size(); i++) {
+                        PaperAPITools.instance.setSignLine(sign, i, list.get(i));
+                    }
+                }
+                state.update();
             }
-            else {
-                for (int i = 0; i < list.size(); i++) {
-                    PaperAPITools.instance.setSignLine(state, i, list.get(i));
+        }
+
+        // <--[mechanism]
+        // @object LocationTag
+        // @name sign_sides_contents
+        // @input MapTag
+        // @description
+        // Sets the contents of each side for a sign block.
+        // The input map keys are either 'Front' or 'Back'
+        // @tags
+        // <LocationTag.sign_contents>
+        // -->
+        if (mechanism.matches("sign_sides_contents") && mechanism.requireObject(MapTag.class)) {
```

Comment:
New mechanisms should be added using the modern `registerMechanism`.
Also instead of a map with keys for both sides, might be easier to understand as a mech+tag pair for the front (which already exists) and a mech+tag pair for the back? I.e. something like `sign_back_contents`